### PR TITLE
Restore `jj bench` commands, and add new ones for revsets

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: src/commands/bench.rs [hunk 1]
```diff
@@ -149,6 +157,44 @@ pub(crate) fn cmd_bench(
...
+            let file_path = command.cwd().join(&command_matches.file);
+            let revsets = std::fs::read_to_string(&file_path)?;
+            let mut criterion = Criterion::default();
+            let mut group = criterion.benchmark_group("revsets");
+            for revset in revsets.split('\n') {
+                let revset = revset.trim();
+                if revset.starts_with('#') || revset.is_empty() {
+                    continue;
+                }
+                ui.write(&format!("----------Testing revset: {revset}----------\n"))?;
```
Comment (yuja) [inline src/commands/bench.rs:171]: nit: `writeln!(ui, ...)`

**Review by yuja — APPROVED** [review]

File: src/commands/bench.rs [hunk 2]
```diff
@@ -149,6 +157,44 @@ pub(crate) fn cmd_bench(
                 routine,
             )?;
         }
+        BenchCommands::Revsets(command_matches) => {
+            let workspace_command = command.workspace_helper(ui)?;
+            let file_path = command.cwd().join(&command_matches.file);
+            let revsets = std::fs::read_to_string(&file_path)?;
+            let mut criterion = Criterion::default();
+            let mut group = criterion.benchmark_group("revsets");
+            for revset in revsets.split('\n') {
```
[the remark this answers is not in the document]
Comment (martinvonz) [reply src/commands/bench.rs]: Done

File: src/commands/bench.rs (hunk 1 above)
Comment (martinvonz) [reply src/commands/bench.rs]: Done

File: src/commands/bench.rs [hunk 3]
```diff
@@ -126,36 +124,31 @@ pub(crate) fn cmd_bench(
...
+            // Time both evaluation and iteration
             let routine = || {
-                index
-                    .walk_revs(
-                        &[wanted_commit.id().clone()],
-                        &[unwanted_commit.id().clone()],
-                    )
-                    .count()
+                let count = workspace_command
+                    .evaluate_revset(expression.clone())
```
[the remark this answers is not in the document]
Comment (yuja) [reply src/commands/bench.rs:150]: Perhaps `iter_batched()` can be used. It runs `(0..batch_size).map(|_| setup()).collect()` before starting measurement. I'll take a look.

https://docs.rs/criterion/latest/criterion/struct.Bencher.html#method.iter_batched

**Pull request merged**